Automattic/wp-calypso
# Add more generic ProductIcon component that is going to replace PlanIcon

This PR introduces a `<ProductIcon />` - a more generic version of `<PlanIcon />` component that supports not only plans but also products icons. It's needed in the new `<PurchasesListing />` component to support Jetpack Backup (#37727).

![Screen Shot on 2019-11-20 at 13-21-55](https://user-images.githubusercontent.com/478735/69238650-c2d69a00-0b98-11ea-81f6-515ed446913a.png)

The current plan is to replace all occurrences of `<PlanIcon />` with this new component in the subsequent PR.

#### Changes proposed in this Pull Request

* Introduce a `<ProductIcon />` that's based and is meant to replace the `<PlanIcon />` component

#### Testing instructions



* Go to http://calypso.localhost:3000/devdocs/design/product-icon
* Confirm you see all sorts of icons (plans, Jetpack plans and Jetpack products)

Fixes n/a
